### ISSUE TITLE
Unmet dependencies, incompatible with Windows?

### ISSUE THREAD

**Issue by mcurrington** [gatsbyjs/gatsby#19122]
## Description

Dependency warnings on basic site creation. This may not be a bug and related to my environment but I am not sure?

### Steps to reproduce

Whilst installing gatsby I am getting these dependency warnings. 

$ gatsby new basic-site
info Creating new site from git: https://github.com/gatsbyjs/gatsby-starter-defa                                                        ult.git
Cloning into 'basic-site'...
remote: Enumerating objects: 24, done.
remote: Counting objects: 100% (24/24), done.
remote: Compressing objects: 100% (16/16), done.
remote: Total 2299 (delta 13), reused 19 (delta 8), pack-reused 2275
Receiving objects: 100% (2299/2299), 12.41 MiB | 1.90 MiB/s, done.
Resolving deltas: 100% (1479/1479), done.
success Created starter directory layout
info Installing packages...
yarn install v1.19.1
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.1.1: The platform "win32" is incompatible with this module.
info "fsevents@2.1.1" is an optional dependency and failed compatibility check.                                                         Excluding it from installation.
info fsevents@1.2.9: The platform "win32" is incompatible with this module.
info "fsevents@1.2.9" is an optional dependency and failed compatibility check.                                                         Excluding it from installation.
[3/4] Linking dependencies...
warning "gatsby > @typescript-eslint/eslint-plugin > tsutils@3.17.1" has unmet p                                                        eer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-d                                                        ev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-                                                        beta".
[4/4] Building fresh packages...
Done in 47.43s.
info Initialising git in basic-site
Initialized empty Git repository in C:/cygwin64/home/Matt/basic-site/.git/
info Create initial git commit in basic-site
info
Your new Gatsby site has been successfully bootstrapped. Start developing it by                                                         running:
  $ cd basic-site
  $ gatsby develop

### Expected result

The site should build cleanly.

### Actual result

It was built with warnings.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.

  System:
    OS: Windows 10
    CPU: (4) x64 Intel(R) Core(TM) i5-4300U CPU @ 1.90GHz
  Binaries:
    Yarn: 1.19.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.11.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 44.18362.387.0
  npmPackages:
    gatsby: ^2.17.6 => 2.17.6 
    gatsby-image: ^2.2.30 => 2.2.30 
    gatsby-plugin-manifest: ^2.2.25 => 2.2.25 
    gatsby-plugin-offline: ^3.0.17 => 3.0.17 
    gatsby-plugin-react-helmet: ^3.1.13 => 3.1.13 
    gatsby-plugin-sharp: ^2.2.34 => 2.2.34 
    gatsby-source-filesystem: ^2.1.35 => 2.1.35 
    gatsby-transformer-sharp: ^2.3.1 => 2.3.1

**Comment by pieh** [comment]
The `info fsevents@2.1.1: The platform "win32" is incompatible with this module.` is about dependency specific to MacOS in one of upstream packages we use. It has alternative implementations for other platforms, just this can't be expressed in dependency declarations so `npm` will show this message. It's declared as optional dependency so installation still is successful. There's just nothing we can do about this confusing `npm` message.

Unmet dependency warning is similar case where one of our dependency list `typescript` as peerDependency but really it's optional peer dependency. This is something that we potentially could ask the other package to sort out to avoid this confusing message, but still - this won't affect your installation

Long story short - those warnings won't affect usage of gatsby on Windows

**Comment by sidharthachatterjee** [comment]
Marking this answered and closing for now. Thanks!
